phoenix-core/src/it/java/org/apache/phoenix/end2end/OrphanChildLinkRowsIT.java

```diff
@@ -0,0 +1,218 @@
+     */
+    @Test
+    public void testChildLinkQueryWithLimit() throws Exception {
+
+        ConnectionQueryServicesTestImpl.setFailPhaseThreeChildLinkWriteForTesting(true);
+        ChildLinkScanTask.disableChildLinkScanTask(true);
+
+        String CREATE_TABLE_DDL = "CREATE TABLE %s (TENANT_ID VARCHAR NOT NULL, A INTEGER NOT NULL, B INTEGER CONSTRAINT PK PRIMARY KEY (TENANT_ID, A))";
+        String CREATE_VIEW_DDL = "CREATE VIEW %s (NEW_COL1 INTEGER, NEW_COL2 INTEGER) AS SELECT * FROM %s WHERE B > 10";
+
+        try (Connection connection = DriverManager.getConnection(getUrl())) {
+            for (int i=0; i<10; i++) {
+                String table1 = "T_" + generateUniqueName();
+                String table2 = "T_" + generateUniqueName();
+                String view = "V_" + generateUniqueName();
+                connection.createStatement().execute(String.format(CREATE_TABLE_DDL, table1));
+                connection.createStatement().execute(String.format(CREATE_TABLE_DDL, table2));
+                connection.createStatement().execute(String.format(CREATE_VIEW_DDL, view, table1));
+                expectedChildLinks.put(table1, view);
+                try {
+                    connection.createStatement().execute(String.format(CREATE_VIEW_DDL, view, table2));
+                }
+                catch (TableAlreadyExistsException e) {
+
+                }
+            }
+
+            String childLinkQuery = "SELECT * FROM SYSTEM.CHILD_LINK LIMIT 7";
+            ResultSet rs = connection.createStatement().executeQuery(childLinkQuery);
+            int count = 0;
+            while (rs.next()) {
+                count++;
+            }
+            Assert.assertEquals("Incorrect number of child link rows returned", 7, count);
+        }
+    }
+
+    private void verifyNoOrphanChildLinkRow() throws Exception {
+        String childLinkQuery = "SELECT * FROM SYSTEM.CHILD_LINK";
+        try (Connection connection = DriverManager.getConnection(getUrl())) {
+            ResultSet rs = connection.createStatement().executeQuery(childLinkQuery);
+            int count = 0;
+            while (rs.next()) {
+                String parentFullName = SchemaUtil.getTableName(rs.getString(TABLE_SCHEM), rs.getString(TABLE_NAME));
+                Assert.assertTrue("Found Orphan Child Link: " + parentFullName + "->" + rs.getString(COLUMN_FAMILY), expectedChildLinks.containsKey(parentFullName));
+                Assert.assertEquals(String.format("Child was not correct in Child Link. Expected : %s, Actual: %s", expectedChildLinks.get(parentFullName), rs.getString(COLUMN_FAMILY)),
+                        expectedChildLinks.get(parentFullName), rs.getString(COLUMN_FAMILY));
+                count++;
+            }
+            Assert.assertTrue("Found Orphan Linking Row", count <= expectedChildLinks.size());
+            Assert.assertTrue("All expected Child Links not returned by query", count >= expectedChildLinks.size());
```

Comment:
These assert checks are confusing. Isn't it equivalent to `count == expectedChildLinks.size()` How can both be true ?

Reply:
yes it is equivalent, I just wanted to differentiate between 2 cases: 
1) if child_link had more rows than expected, that means there are orphans. 
2) if child_link returned fewer rows than expected, something went wrong with the query.